prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java

```diff
@@ -503,20 +504,34 @@ public int read() throws IOException {
         return getBufferedRequest(new HttpServletRequestInputStreamWrapper(request, servletInputStream));
     }
 
-
-    private void validateFeeds(List<Entity> feeds) throws FalconException {
+    private void validateFeeds(List<Entity> feeds, String jobName) throws FalconException {
         for (Entity feed : feeds) {
+            checkIfPartOfAnotherExtension(feed, EntityType.FEED, jobName);
             super.validate(feed);
         }
     }
 
-    private void validateProcesses(List<Entity> processes) throws FalconException {
+    private void validateProcesses(List<Entity> processes, String jobName) throws FalconException {
         ProcessEntityParser processEntityParser = new ProcessEntityParser();
         for (Entity process : processes) {
+            checkIfPartOfAnotherExtension(process, EntityType.PROCESS, jobName);
             processEntityParser.validate((Process) process, false);
         }
     }
 
+    private void checkIfPartOfAnotherExtension(Entity process, EntityType entityType, String jobName)
+        throws FalconException {
+        try {
+            Entity entity = EntityUtil.getEntity(entityType, process.getName());
+            String extractedJobName = AbstractExtensionManager.getJobNameFromTag(entity.getTags());
+            if (!extractedJobName.equals(jobName)) {
+                throw new FalconException("Entity is part another extension job: " + extractedJobName);
```

Comment:
Message could be slightly misleading. If extractedJobName is null (when the entity is not part of a Job), this will still say it is part of an extension.